web/src/upload.ts

```diff
@@ -372,13 +372,14 @@ export function setup_upload(config: Config): Uppy {
 
     uppy.on("upload-success", (file, response) => {
         assert(file !== undefined);
-        const {url} = z.object({url: z.string().optional()}).parse(response.body);
+        const {url, filename} = z
+            .object({url: z.string().optional(), filename: z.string().optional()})
+            .parse(response.body);
         if (url === undefined) {
             return;
         }
```

Comment:
Are url/filename actually optional in this API? I suspect that this is confused error-handling code, and we just want to be doing `.safeParse` and returning with a blueslip error if there's a parse failure (which probably means `uppy` sent this function something that wasn't an API response).